src/main/java/org/sopt/service/PostService.java

```diff
@@ -0,0 +1,90 @@
+package org.sopt.service;
+
+import org.sopt.domain.Post;
+import org.sopt.repository.PostRepository;
+
+import java.io.*;
+import java.time.LocalDateTime;
+import java.time.Duration;
+import java.util.List;
+
+public class PostService {
+    private final PostRepository postRepository = new PostRepository();
+
+    public boolean createPost(String title) {
+        boolean flag = true;
+
+        if (!this.getAllPosts().isEmpty()) {
+            LocalDateTime latestPostTime = getAllPosts().get(getAllPosts().size() - 1).getTime();
+            LocalDateTime now = LocalDateTime.now();
+            Duration duration = Duration.between(latestPostTime, now);
+            if (duration.toMinutes() < 3) {
+                flag = false;
+            }
+        }
```

Comment:
다진님 말씀대로 중복된 `getAllPosts()` 호출을 피해보는 방향으로 리팩터링 해주면 더 깔끔한 코드가 될 것 같아요 ! 

현재 흐름은 조건문 내에서 `getAllPosts()` 1회 호출 > `getAllPosts().get(getAllPosts().size() - 1).getTime();` 내에서 `getAllPosts()`를 2회 호출 하는 구조입니다.

```suggestion
List<Post> posts = this.getAllPosts();
if (!posts.isEmpty()) {
    Post latestPost = posts.get(posts.size() - 1); // 가장 최근 게시물
    LocalDateTime latestPostTime = latestPost.getTime();
    Duration duration = Duration.between(latestPostTime, LocalDateTime.now());
    if (duration.toMinutes() < 3) {
        flag = false;
    }
}
```

위와 같이 1회 호출로 가져온 `posts` 를 가지고 로직을 처리해주는 것이 조금 더 나아보인다고 생각합니다 !